server/webapp/WEB-INF/rails.new/app/controllers/api_v1/users_controller.rb

```diff
@@ -43,12 +37,48 @@ def update
       end
     end
 
+    def create
+      result = HttpLocalizedOperationResult.new
+
+      user, created = user_service.withEnableUserMutex do
```

Comment:
Because of tristate? Anyway, I'm sure you've thought about it more than I have. My question was based on how a user is created, when called [from the admin page](https://github.com/gocd/gocd/blob/master/server/webapp/WEB-INF/rails.new/app/controllers/admin/users_controller.rb#L29-L40).